core/src/main/java/hudson/util/RobustReflectionConverter.java

```diff
@@ -307,9 +308,11 @@ public Object doUnmarshal(final Object result, final HierarchicalStreamReader re
                         implicitCollectionsForCurrentObject = writeValueToImplicitCollection(context, value, implicitCollectionsForCurrentObject, result, fieldName);
                     }
                 }
+            } catch (CriticalXStreamException e) {
+                throw e;
             } catch (XStreamException e) {
                 if (critical) {
-                    throw e;
+                    throw new CriticalXStreamException(e);
```

Comment:
I don't think I get your point. Please let me know more details.
`CriticalXStreamException` wraps only `XStreamException` for its constructor.

What I want to do is to let callers know that the error is critical.